code/modules/mob/living/simple_animal/slaughter/slaughter.dm

```diff
@@ -0,0 +1,433 @@
+					else
+						victim.loc = holder///holder
+						src.visible_message("<span class='danger'><B>The [src] drags [victim] into [B]!</B></span>")
+						src.kidnapped = victim
+			flick("jaunt",animation)
+			src.loc = holder
+			src.phased = TRUE
+			src.holder = holder
+
+			if(src.kidnapped)
+				src << "<B>You begin to feast on [kidnapped]. You can not move while you are doing this.</B>"
+				src.visible_message("<span class='warning'><B>Loud eating sounds come from the blood...</B></span>")
+				src.eating = TRUE
+				sleep(6)
+				qdel(animation)
+				src.GibEat()
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				src.GibEat()
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				src.GibEat()
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				src << "<B>You devour [kidnapped]. Your health is fully restored.</B>"
+				src.adjustBruteLoss(-1000)
+				kidnapped.ghostize()
+				qdel(kidnapped)
+				src.devoured++
+				//src << "<B> [src.devoured] [src.kidnapped] [src.eating]<B>"//Debug for tracking how many is devoured, use in objective?
+				src.kidnapped = null
+				src.eating = FALSE
+				src.pulling = FALSE
+				if (src.devoured == 5)
+					src.mind.current.verbs += /mob/living/simple_animal/slaughter/proc/goreThrow
+					src.mind.current.verbs += /mob/living/simple_animal/slaughter/proc/bloodSac
+					src << "<span class='notice'> You have consumed enough to be able to summon Excess Gore."
+			src.notransform = 0
+			if(!(src.eating))
+				new /obj/effect/gibspawner/human(get_turf(src))///Somewhere a janitor weeps..
+				sleep(6)
+				qdel(animation)
+
+/mob/living/simple_animal/slaughter/proc/phasein(var/obj/effect/decal/cleanable/B)
+	var/atom/movable/overlay/animation = new /atom/movable/overlay( B.loc )
+	animation.name = "odd blood"
+	animation.density = 0
+	animation.anchored = 1
+	animation.icon = 'icons/mob/mob.dmi'
+	animation.icon_state = "jauntup" //Paradise Port:I reversed the jaunt animation so it looks like its rising up
+	animation.layer = 5
+	animation.master = B.loc
+	animation.dir = src.dir
+	if(src.eating)
+		src << "<B>Finish eating first!</B>"
+	else
+		flick("jauntup",animation)
+		src.client.eye = src
+		src.loc = B.loc
+		src.phased = FALSE
+		//Paradise port:The hallucination sounds removed from the demon as of a recent commit..i liked it though..
+		if (prob(25))
+			var/list/voice = list('sound/hallucinations/behind_you1.ogg','sound/hallucinations/im_here1.ogg','sound/hallucinations/turn_around1.ogg','sound/hallucinations/i_see_you1.ogg')
+			playsound(get_turf(src), pick(voice),50, 1, -1)
+		src.visible_message("<span class='danger'><B>The [src] rises out of [B]!</B></span>")
+		new /obj/effect/gibspawner/human(get_turf(src))
+		playsound(get_turf(src), 'sound/misc/exit_blood.ogg', 100, 1, -1)
+		sleep(6)
+		qdel(src.holder)
```

Comment:
Both of these qdels should make sure what they are attempting to qdel actually exists.